src/renderer/gl_engine/tvgGlTessellator.cpp

```diff
@@ -1707,15 +1707,26 @@ void Stroker::doDashStroke(const PathCommand *cmds, uint32_t cmd_count, const Po
 
 void Stroker::strokeCap()
 {
-    if (mStrokeState.firstPt == mStrokeState.prevPt) return;
     if (mStrokeCap == StrokeCap::Butt) return;
 
     if (mStrokeCap == StrokeCap::Square) {
-        strokeSquare(mStrokeState.firstPt, {-mStrokeState.firstPtDir.x, -mStrokeState.firstPtDir.y});
-        strokeSquare(mStrokeState.prevPt, mStrokeState.prevPtDir);
+        if (mStrokeState.firstPt == mStrokeState.prevPt) {
+            if (mStrokeWidth > 0) {
+                strokeSquarePoint(mStrokeState.firstPt);
```

Comment:
This may result in only containing move instructions resulting in drawing StrokeCaps.
Can you help confirm  the rendering result like this ?
```
auto path = tvg::Shape::gen();

path->moveTo(100, 100);
path->moveTo(150, 100);
path->moveTo(200, 100);
path->strokeWidth(10);
path->strokeJoin(tvg::StrokeJoin::Round);
```

Reply:
@RuiwenTang Good point. In the proposed solution, caps were drawn even for move operations, which was not consistent with what the SW renderer does. The code above was drawing three points, which should not happen. I have updated my pull request with what I believe is a better handling of this scenario, ensuring it aligns with the behavior of other renderers.